packages/h5wasm/src/h5wasm-api.test.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  assertDataset,
+  assertGroup,
+  assertGroupWithChildren,
+  hasNonNullShape,
+} from '@h5web/shared/guards';
+import type {
+  ArrayShape,
+  Dataset,
+  ScalarShape,
+} from '@h5web/shared/hdf5-models';
+import { existsSync } from 'fs';
+import { readFile } from 'fs/promises';
+import path from 'path';
+import { expect, test } from 'vitest';
+
+import { H5WasmApi } from './h5wasm-api';
+
+const LOCAL_TEST_FILE = path.resolve(__dirname, '../dist-h5/sample.h5');
+const REMOTE_TEST_FILE = 'http://www.silx.org/pub/h5web/sample.h5'; // `https` would complicate things...
+
+async function loadTestFile(): Promise<ArrayBuffer> {
+  if (existsSync(LOCAL_TEST_FILE)) {
+    return readFile(LOCAL_TEST_FILE);
+  }
+
+  const resp = await fetch(REMOTE_TEST_FILE);
+  return resp.arrayBuffer();
+}
```

Comment:
The test first looks for the sample file at `packages/h5wasm/dist-h5/sample.h5`, as this is where the Python script creates it. If the sample file hasn't been generated locally, then the test will fetch it from silx.org.